In [8]:
import numpy as np
from scipy.stats import norm

## Z-test

Reported average credit card debt for college students is $\mu=3262$.
Survey of $n=50$ students found their average debt as $\overline{X}=2995$.
Population standard deviation is $\sigma = 1100$.


$H_0$: $\mu = 3262$

$H_1$: $\mu < 2995$ (left-tailed z-test)

The test statistics $z$ is

$$
z = \frac{\overline{X} - \mu}{\frac{\sigma}{\sqrt{n}}}
$$

and follows the normal distribution under the null hypothesis.

In [6]:
def z_test(mean_x, mu, sigma, n):
    return (mean_x - mu) / (sigma / np.sqrt(n)) 

In [14]:
z = z_test(mean_x=2995, mu=3262, sigma=1100, n=50)
print(z)

-1.7163410052437107


Our test statistics is

$$
z = -1.7163
$$

## Using Critical Values

With a chosen significance level $\alpha = 0.05$,
we want to find a critical region $[-\infty, c]$ (left-tailed test) such that

$$
\int_{-\infty}^{c}p(x)\mathrm{d}x = \alpha = 0.05
$$

where $p(x)$ is a $\mathcal{N}(0, 1)$ probability density function.
We can find the critical value $c$ defining the region using 
the inverse of the normal CDF (cumulative density function).

In [12]:
# ppf is cdf inverse
c = norm.ppf(0.05)

# just to be sure
print(c, norm.cdf(c))

-1.6448536269514729 0.049999999999999975


With $z = -1.7163$ and $c = -1.6445$, __we reject the null hypothesis__ as
$z < c$ at $\alpha = 0.05$ significance level.

## Using p-values

The p-value for our statistics can be found using the CDF.

In [20]:
p = norm.cdf(z)
print(p)

0.04304982368670932


We reject the null hypothesis as $p < \alpha$
at $\alpha = 0.05$ significance level.